src/tests/ftest/pool/rebuild_with_io.py

```diff
@@ -20,143 +20,93 @@
-            while True:
-                pool.pool_query()
-                if pool.pool_info.pi_rebuild_st.rs_done == 1:
-                    break
-                else:
-                    time.sleep(2)
-
-            # check rebuild statistics
-            if pool.pool_info.pi_ndisabled != 1:
-                self.fail("Number of disabled targets reporting incorrectly: {}"
-                          .format(pool.pool_info.pi_ndisabled))
-            if pool.pool_info.pi_rebuild_st.rs_errno != 0:
-                self.fail("Rebuild error reported: {}".format(
-                    pool.pool_info.pi_rebuild_st.rs_errno))
-            if pool.pool_info.pi_rebuild_st.rs_obj_nr <= 0:
-                self.fail("No objects have been rebuilt.")
-            if pool.pool_info.pi_rebuild_st.rs_rec_nr <= 0:
-                self.fail("No records have been rebuilt.")
-
-        except (ValueError, DaosApiError) as excep:
-            print(excep)
-            print(traceback.format_exc())
-            self.fail("Expecting to pass but test has failed.\n")
-
-        finally:
-            # wait for the I/O process to finish
-            try:
-                server_utils.stop_server(hosts=self.hostlist_servers)
-                os.remove(hostfile_servers)
-                # really make sure everything is gone
-                check_for_pool.cleanup_pools(self.hostlist_servers)
-            finally:
-                if self.agent_sessions:
-                    agent_utils.stop_agent(self.agent_sessions)
-                server_utils.kill_server(self.hostlist_servers)
+        # Get the test params
+        pool = TestPool(self.context, self.log)
+        pool.get_params(self)
+        container = TestContainer(pool)
+        container.get_params(self)
+        targets = self.params.get("targets", "/run/server_config/*")
+        # data = self.params.get("datasize", "/run/testparams/*")
+        rank = self.params.get("rank", "/run/testparams/*")
+        obj_class = self.params.get("object_class", "/run/testparams/*")
+        server_count = len(self.hostlist_servers)
+
+        # Create a pool and verify the pool info before rebuild (also connects)
+        pool.create()
+        status = pool.check_pool_info(
+            pi_nnodes=server_count,
+            pi_ntargets=(server_count * targets),  # DAOS-2799
+            pi_ndisabled=0,
+        )
+        status &= pool.check_rebuild_status(
+            rs_done=1, rs_obj_nr=0, rs_rec_nr=0, rs_errno=0)
+        self.assertTrue(status, "Error confirming pool info before rebuild")
+
+        # Create and open the contaner
+        container.create()
+
+        # Write data to the contianer for 30 seconds
+        self.log.info(
+            "Wrote %s bytes to container %s",
+            container.execute_io(30, rank, obj_class), container.uuid)
+
+        # Determine how many objects will need to be rebuilt
+        container.get_target_rank_list_data(rank, " prior to rebuild")
```

Comment:
(pylint-no-member) Instance of 'TestContainer' has no 'get_target_rank_list_data' member